test/Generators/Microsoft.Gen.BuildMetadata/Unit/GeneratorTests.cs

```diff
@@ -89,12 +90,13 @@ private static TestAnalyzerConfigOptionsProvider CreateTestOptionsProvider(bool?
         // Create the generator driver with the options provider
         var driver = Microsoft.CodeAnalysis.CSharp.CSharpGeneratorDriver.Create(
             generators: new[] { new BuildMetadataGenerator().AsSourceGenerator() },
+            parseOptions: CSharpParseOptions.Default.WithLanguageVersion(LanguageVersion.Preview),
             optionsProvider: optionsProvider);
 
-        var result = driver.RunGenerators(comp!);
+        var result = driver.RunGeneratorsAndUpdateCompilation(comp!, out var outputCompilation, out var diagnostics);
```

Comment:
`diagnostics` is just the diagnostics your generator would have reported itself; the `global::System.string` issue would not have been caught by this. Please verify that `compilation.GetDiagnostics` either returns nothing, or the set of diagnostics you're expecting.